python/lsst/ts/standardscripts/enable_group.py

```diff
@@ -0,0 +1,87 @@
+# This file is part of ts_standardscripts
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["EnableGroup"]
+
+import abc
+
+from lsst.ts import salobj
+
+
+class EnableGroup(salobj.BaseScript, metaclass=abc.ABCMeta):
+    """Base Script for enabling groups of CSCs.
+
+    Parameters
+    ----------
+    index : `int`
+        Index of Script SAL component.
+
+    Notes
+    -----
+    **Checkpoints**
+
+    **Details**
+
+    """
+
+    __test__ = False  # stop pytest from warning that this is not a test
+
+    def __init__(self, index, descr):
+        super().__init__(index=index, descr=descr)
+
+        self.config = None
+
+    @property
+    @abc.abstractmethod
+    def group(self):
+        """Return group of CSC attribute.
+        """
+        raise NotImplementedError()
+
+    async def configure(self, config):
+        self.config = config
+
+    def set_metadata(self, metadata):
+        metadata.duration = 60.0
+
+    async def run(self):
+        if hasattr(self.config, "ignore"):
+            for comp in self.config.ignore:
+                rname = comp.lower().replace(":", "_")
```

Comment:
I strongly suggest using the same name for ignore and group.components. It should greatly reduce confusion. Any time there is a mismatch you have to document the heck out of it, and folks are likely to still find themselves wondering which version of the name to use where.

I don't think the colon will be a problem for yaml. As long as the colon is embedded in the name, yaml seems to handle it just fine:
```
>>> yaml.safe_load("foo:bar: 1")
{'foo:bar': 1}
```
If you insist on remapping names then please document it here and in the schemas for the subclasses of this class.

Reply:
good point..